scipy/optimize/_minimize.py

```diff
@@ -581,50 +582,79 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
-        return _minimize_bfgs(fun, x0, args, jac, callback, **options)
+        res = _minimize_bfgs(fun, x0, args, jac, callback, **options)
     elif meth == 'newton-cg':
-        return _minimize_newtoncg(fun, x0, args, jac, hess, hessp, callback,
+        res = _minimize_newtoncg(fun, x0, args, jac, hess, hessp, callback,
                                   **options)
     elif meth == 'l-bfgs-b':
-        return _minimize_lbfgsb(fun, x0, args, jac, bounds,
+        res = _minimize_lbfgsb(fun, x0, args, jac, bounds,
                                 callback=callback, **options)
     elif meth == 'tnc':
-        return _minimize_tnc(fun, x0, args, jac, bounds, callback=callback,
+        res = _minimize_tnc(fun, x0, args, jac, bounds, callback=callback,
                              **options)
     elif meth == 'cobyla':
-        return _minimize_cobyla(fun, x0, args, constraints, **options)
+        res = _minimize_cobyla(fun, x0, args, constraints, **options)
     elif meth == 'slsqp':
-        return _minimize_slsqp(fun, x0, args, jac, bounds,
+        res = _minimize_slsqp(fun, x0, args, jac, bounds,
                                constraints, callback=callback, **options)
     elif meth == 'trust-constr':
-        return _minimize_trustregion_constr(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trustregion_constr(fun, x0, args, jac, hess, hessp,
                                             bounds, constraints,
                                             callback=callback, **options)
     elif meth == 'dogleg':
-        return _minimize_dogleg(fun, x0, args, jac, hess,
+        res = _minimize_dogleg(fun, x0, args, jac, hess,
                                 callback=callback, **options)
     elif meth == 'trust-ncg':
-        return _minimize_trust_ncg(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trust_ncg(fun, x0, args, jac, hess, hessp,
                                    callback=callback, **options)
     elif meth == 'trust-krylov':
-        return _minimize_trust_krylov(fun, x0, args, jac, hess, hessp,
+        res = _minimize_trust_krylov(fun, x0, args, jac, hess, hessp,
                                       callback=callback, **options)
     elif meth == 'trust-exact':
-        return _minimize_trustregion_exact(fun, x0, args, jac, hess,
+        res = _minimize_trustregion_exact(fun, x0, args, jac, hess,
                                            callback=callback, **options)
     else:
         raise ValueError('Unknown solver %s' % method)
 
+    # cannot be a successful minimisation if the bounds are exceeded, or
+    # constraints aren't met
+    if bounds is not None:
+        # force to be new-style
+        b = standardize_bounds(bounds, res.x, 'trust-constr')
+        if np.any(res.x > b.ub) or np.any(res.x < b.lb):
+            res.success = False
+
+    if constraints is not None and len(constraints):
+        catol = 0.
+        if meth == 'cobyla':
+            catol = 2e-4
```

Comment:
Maybe include a small comment saying what catol is and that 2e-4 is it default value. Should make it easier for someone reading the code. Otherwise  2e-4 just seem arbitrary.